src/overlays/actors/ovl_Obj_Hana/z_obj_hana.c

```diff
@@ -27,11 +26,74 @@ const ActorInit Obj_Hana_InitVars = {
     (ActorFunc)ObjHana_Update,
     (ActorFunc)ObjHana_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Hana/ObjHana_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Hana/ObjHana_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 8, 10, 0, { 0, 0, 0 } },
+};
+
+static CollisionCheckInfoInit sColChkInfoInit = { 0, 0xC, 0x3C, 0xFF };
+
+typedef struct {
+    /* 0x00 */ Gfx* dList;
+    /* 0x04 */ f32 scale;
+    /* 0x08 */ f32 yOffset;
+    /* 0x0C */ s16 radius;
+    /* 0x0E */ s16 height;
+} HanaParams; // size 0x10
+
+static HanaParams hanaParams[] = {
+    { 0x05000500, 0.01f, 0.0f, -1, 0 },
+    { 0x0500A880, 0.1f, 58.0f, 10, 18 },
+    { 0x0500B9D0, 0.4f, 0.0f, 12, 44 },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 10, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 900, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 60, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
+};
+
+void ObjHana_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHana* this = THIS;
+    s16 type = this->actor.params & 3;
+    HanaParams* params = &hanaParams[type];
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Actor_SetScale(&this->actor, params->scale);
+    this->actor.shape.unk_08 = params->yOffset;
+    if (params->radius >= 0) {
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        this->collider.dim.radius = params->radius;
+        this->collider.dim.height = params->height;
+        func_80061ED4(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    }
+
+    if (type == 2 && (gSaveContext.eventChkInf[4] & 1) != 0) {
```

Comment:
Flag checks are treated as boolean checks for which we omit the `!= 0` or `== 0`
```suggestion
    if (type == 2 && (gSaveContext.eventChkInf[4] & 1)) {
```